tests/bls_tests.rs

```diff
@@ -0,0 +1,297 @@
+#![cfg(feature = "spec_tests")]
+
+use ethereum_consensus::crypto::{
+    aggregate, aggregate_verify, fast_aggregate_verify, PublicKey, SecretKey, Signature,
+};
+use glob::glob;
+use hex;
+use serde::{de::Error, Deserialize, Deserializer, Serialize};
+use serde_yaml;
+use std::fmt::Debug;
+use std::fs::File;
+
+fn decode_hex_with_prefix<'a, D>(input: D) -> Result<Vec<u8>, D::Error>
+where
+    D: Deserializer<'a>,
+{
+    let data: String = Deserialize::deserialize(input)?;
+    let data = data.as_bytes();
+    let data = if data.starts_with(b"0x") {
+        &data[2..]
+    } else {
+        data
+    };
+    //null character case, empty array
+    if data == [126] {
+        return Ok(vec![]);
+    }
+    hex::decode(data).map_err(D::Error::custom)
+}
+fn decode_hexvec_with_prefix<'a, D>(input: D) -> Result<Vec<Vec<u8>>, D::Error>
```

Comment:
nitpick, but i'd suggest an extra `_` here:
```suggestion
fn decode_hex_vec_with_prefix<'a, D>(input: D) -> Result<Vec<Vec<u8>>, D::Error>
```